### PR TITLE
Add support for slash commands

### PR DESCRIPTION
**Pull request by V13Axel**
This addresses #227 and allows the use of Slack slash commands, as below.

Simply prepend `/slack slash` to whatever you'd run via the official clients, and your command will be run. Example:

`/command arg1 arg2 arg3` simply becomes `/slack slash /command arg1 arg2 arg3`.

I've found it helpful to create an alias of `/command` => `/slack slash command` via weechat for the ones I use.

### CODE REVIEW

File: wee_slack.py [hunk 1]
```diff
@@ -1308,6 +1308,30 @@ def command_markread(current_buffer, args):
...
+@slack_buffer_required
+def command_slash(current_buffer, args):
+    """
+    Support for custom slack commands
+    /slack slash /customcommand arg1 arg2 arg3
+    """
+    aargs = args.split(None, 1)
+
+    if len(aargs) == 0:
+        # XXX: whine
```
Comment (tfheen) [inline wee_slack.py]: We should offer a useful error in this case.

File: wee_slack.py [hunk 2]
```diff
@@ -1308,6 +1308,30 @@ def command_markread(current_buffer, args):
...
+        return
+    command = aargs[0]
+    text = aargs[1]
+
+    channel = current_buffer_name(short=True)
+    domain = current_domain_name()
+
+    if servers.find(domain).channels.find(channel):
+        channel_identifier = servers.find(domain).channels.find(channel).identifier
```
Comment (tfheen) [inline wee_slack.py]: What happens if we don't find the channel?

**Comment by V13Axel** [comment]
@tfheen How's that look? Cleaned things up a bit and addressed both of your concerns (As well as an error for just typing `/slack slash /command` with no args)

File: wee_slack.py [hunk 3]
```diff
@@ -1308,6 +1308,37 @@ def command_markread(current_buffer, args):
...
+    Support for custom slack commands
+    /slack slash /customcommand arg1 arg2 arg3
+    """
+
+    server = servers.find(current_domain_name())
+    channel = current_buffer_name(short=True)
+    domain = current_domain_name()
+
+    if args is None:
+        server.buffer_prnt("Usage: /slack slash [/someslashcommand].")
```
Comment (tfheen) [inline wee_slack.py]: This should probably read `"Usage: /slack slash /someslashcommand [arguments...]"`

File: wee_slack.py [hunk 4]
```diff
@@ -1308,6 +1308,37 @@ def command_markread(current_buffer, args):
...
+        return
+
+    aargs = args.split(None, 1)
+
+    command = aargs[0]
+    if 1 < len(aargs):
```
Comment (tfheen) [inline wee_slack.py]: `text = args[1] if len(args > 1) else ""`

**Comment by V13Axel** [comment]
Whoot, +1 for @rawdigits double-checking my testing of it and merging it in :wink:

**Comment by l2ol33rt** [comment]
@tfheen any update on testing this?

I'm running it off @V13Axel's `master` branch and it's awesome!

File: README.md [hunk 5]
```diff
@@ -32,6 +32,7 @@ Features
   * Away/back status handling
   * Expands/shows metadata for things like tweets/links
   * Displays edited messages (slack.com irc mode currently doesn't show these)
+  * Slash commands (including custom ones!)
```
Comment (V13Axel) [inline README.md]: Heh. I had forgotten that existed, will do.

Comment (kenyonj) [inline README.md]: :+1: Excited to get this in!

**Pull request merged**